src/xenia/base/threading_win.cc

```diff
@@ -453,6 +453,15 @@ std::unique_ptr<Thread> Thread::Create(CreationParameters params,
   return std::make_unique<Win32Thread>(handle);
 }
 
+std::unique_ptr<Thread> Thread::GetCurrentThread() {
+  HANDLE handle = ::GetCurrentThread();
+  if (handle == INVALID_HANDLE_VALUE) {
+    return nullptr;
+  }
+
+  return std::make_unique<Win32Thread>(handle);
```

Comment:
it'd be nice to use a `thread_local` class static for this instead - that way there's a 1:1 mapping between Win32Thread and the underlying windows thread (and if GetCurrentThread() is called frequently we aren't thrashing memory. You could still have it allocate on demand (when called from a thread that hadn't been created from this file).